AirCasting/Utils/AveragingService.swift

```diff
@@ -81,39 +78,45 @@ final class AveragingService: NSObject, ObservableObject {
         }
     }
     
-    private func perform(sessionEntity: SessionEntity, averagingWindow: AveragingWindow) {
-        if let uuid = sessionEntity.uuid {
-            measurementStreamStorage.accessStorage { storage in
-                
-                let session = try? storage.getExistingSession(with: uuid)
+    private func perform(uuid: SessionUUID, averagingWindow: AveragingWindow, windowDidChange: Bool) {
+        measurementStreamStorage.accessStorage { storage in
+            let session = try? storage.getExistingSession(with: uuid)
```

Comment:
Calling this function with an uuid of a session that cannot be retrieved from DB seems like a scenario we want to at least take a warning log of. Maybe let's do it instead of `try?`

Reply:
I changed it, now I pass the session to this function and I fetch the session in startPeriodicAveraging.